lib/logstash/pipeline.rb

```diff
@@ -252,6 +252,7 @@ def outputworker
   #
   # This method is intended to be called from another thread
   def shutdown
+    Thread.new { report_inflight_events(@input_to_filter, @filter_to_output, @outputs) }
```

Comment:
I would vote to encapsulate that logic into his own class.
I suggest something like this, so this factory method would take care of the initializing the class and managing the thread.

```
InflightEventsReporter.watch(@input_to_filter, ....)
```